src/rpc/AMMHelpers.cpp

```diff
@@ -0,0 +1,82 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "rpc/AMMHelpers.h"
+
+#include "data/BackendInterface.h"
+#include "util/log/Logger.h"
+
+#include <ripple/protocol/AMMCore.h>
+
+namespace rpc {
+
+std::pair<ripple::STAmount, ripple::STAmount>
+getAmmPoolHolds(
+    BackendInterface const& backend,
+    std::uint32_t sequence,
+    ripple::AccountID const& ammAccountID,
+    ripple::Issue const& issue1,
+    ripple::Issue const& issue2,
+    bool freezeHandling,
```

Comment:
Clio seems not need this flag while rippled uses it.

Reply:
I think @arihantkothari just migrated the code as close as possible to the original implementation inside of `rippled`. It looks like all ends meet though?

Reply:
yea, i believe so